nilearn/decomposition/base.py

```diff
@@ -375,6 +388,20 @@ def fit(self, imgs, y=None, confounds=None):
             self.masker_.fit()
         self.mask_img_ = self.masker_.mask_img_
 
+        # mask_and_reduce step is used only for decomposition estimators i.e.
+        # MultiPCA, CanICA and Dictionary Learning
+        if self.__class__.__name__ != 'BaseDecomposition':
```

Comment:
You don't need this if statement: BaseDecomposition is never used by end user. You can remove the if and simply insert the code that is below.

Reply:
tests will fail because BaseDecomposition has no _raw_fit method implemented.